rules/recording.go

```diff
@@ -188,7 +188,11 @@ func (rule *RecordingRule) HTMLSnippet(pathPrefix string) template.HTML {
 		labels[l.Name] = template.HTMLEscapeString(l.Value)
 	}
 
-	r := rulefmt.Rule{
+	r := struct {
+		Record string            `yaml:"record"`
```

Comment:
Why can't we continue to use rulefmt.Rule here?

Reply:
we want to avoid the yaml.Node here else the output will be a node structure. The node structure is like [this](https://godoc.org/gopkg.in/yaml.v3#Node) which is of course, of no use to us for the htmlsnippet. So for this case where output is string specific and there is no work of debugging details like line/column numbers, we choose to go with normal strings. If we don't do go this way, then the output is like 

```
record:
            kind: 0
            style: 0
            tag: ""
            value: <a href="/test/prefix/graph?g0.expr=testrule&g0.tab=1">testrule</a>
            anchor: ""
            alias: null
            content: []
            headcomment: ""
            linecomment: ""
            footcomment: ""
            line: 0
            column: 0
```

which is of no use here (only).

Reply:
Hmm, so how will we do this once we switch fully to v3?

Reply:
Absolutely not an issue. This format comes only when `yaml.Node` is needed. This was the requirement only for checking the rules and their validation with more control on the details. As far as I see, there seems to be no more use of `yaml.Node` (since non of our components require validation and detailed error in line/column or any other functionalities related to Node), that means a `string` type will work. **`string` type do not have this while unmarshalling**. Only the `yaml.Node` have these details when they are unmarshalled. 

I can submit another PR after this (since this one has a different aim) that converts and unifies the yaml v3 library across Prometheus main repo.

Reply:
So after the followup PR, this anonymous type would be gone?

Reply:
Since both anonymous structs are same, we can rather declare in rulefmt as non yaml node and import from there. Thats what the following will have for this thing.

Reply:
That still doesn't sound right, there shouldn't be anonymous types at all. Otherwise we'll have to duplicate the entire of the config structure just to display it on the UI.

Reply:
How about putting these into a defined structs with string as some types, since node is not required in this case?

Reply:
hm but we already have `rulefmt.Rule{` what's wrong with that? ( I might miss context).

Reply:
My point is that we shouldn't need to duplicate this type at all. We should be able to marshal one type, not have to duplicate types if we want to marshal anything.

Reply:
I think, two types are required, else might look messy, keeping in mind, that we require some features from `v3` and also a marshalled output without `yaml.Node` details.

Reply:
This comment stands.

Reply:
Yes, we can use this now.

Reply:
I think this can be resolved now.